### ISSUE TITLE
rna-transcription test results in segmentation fault

### ISSUE THREAD

**Issue by wolf99** [exercism/xc#116]
In the rna-transcription exercise's test file, the `test_transcription()` test helper function tries to use `strncmp()` and `free()` on the result of the UUT. But, following the TDD rules, at the start of the TDD process the UUT does not allocate any memory - this means that the project is un-buildable until at least some not insignificant portion of the implementation is already achieved, thus it breaks the TDD process that is supposed to be being fostered (according to the description of the exercise at least).

The helper function should be altered so as not to prevent early iterations of the implementation from being built. This could be as simple as adding in a check for `NULL`. e.g:

~~~c
void test_transcription(const char *dna, const char *expected)
{
   char *rna = to_rna(dna);
   if (rna != NULL) {
      TEST_ASSERT_TRUE(strncmp(rna, expected, strlen(dna)) == 0);
      free(rna);
   } else {
      TEST_FAIL();
   }
}
~~~

**Comment by patricksjackson** [comment]
Well, the project can build at least, it will just segfault unhelpfully. I think that's a good idea for a fix, but if you add it please make sure there is at least a brief comment to explain it.

On another note on this piece of code, that `strncmp` call may be wrong. I think the length argument should instead be "`strlen(expected) + 1`" so that it could check if the returned `rna` value has extra garbage characters at the end.

**Comment by wolf99** [comment]
Fixed by #117
